hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ExitCallsEnricher.java

```diff
@@ -0,0 +1,144 @@
+package org.hypertrace.traceenricher.enrichment.enrichers;
+
+import com.google.common.collect.Maps;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.hypertrace.core.datamodel.ApiNodeEventEdge;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.shared.ApiNode;
+import org.hypertrace.core.datamodel.shared.trace.AttributeValueCreator;
+import org.hypertrace.entity.constants.v1.BackendAttribute;
+import org.hypertrace.entity.service.constants.EntityConstants;
+import org.hypertrace.traceenricher.enrichedspan.constants.EnrichedSpanConstants;
+import org.hypertrace.traceenricher.enrichedspan.constants.utils.EnrichedSpanUtils;
+import org.hypertrace.traceenricher.enrichment.AbstractTraceEnricher;
+import org.hypertrace.traceenricher.trace.util.ApiTraceGraph;
+
+public class ExitCallsEnricher extends AbstractTraceEnricher {
+
+  @Override
+  public void enrichTrace(StructuredTrace trace) {
+    Map<ByteBuffer, ApiExitCallInfo> exitCallInfo = computeApiExitCallCount(trace);
+    trace.getEventList().stream()
+        .filter(event -> exitCallInfo.containsKey(event.getEventId()))
+        .forEach(
+            event -> {
+              ApiExitCallInfo apiExitCallInfo = exitCallInfo.get(event.getEventId());
+              addEnrichedAttribute(
+                  event,
+                  EnrichedSpanConstants.API_EXIT_CALLS_ATTRIBUTE,
+                  AttributeValueCreator.create(apiExitCallInfo.getExitCallCount()));
+              addEnrichedAttribute(
+                  event,
+                  EnrichedSpanConstants.API_EXIT_SERVICES_ATTRIBUTE,
+                  AttributeValue.newBuilder()
```

Comment:
I am wondering whether we should enrich callee name to exit calls, or callee id to exit calls. Because, it would be easier to join data using the id, instead of the callee name at the query time (_just like we do it for entity interactions_)

And, also the fact that callee name _could_ potentially be null, for a valid callee id

Reply:
Hey, the callee name won't be null (have added details in your previous comment).

Reply:
As discussed, we are persisting name to count map for now.